Jenkinsfile

```diff
@@ -1,209 +1,211 @@
-                                        }
+pipeline {
+  agent any
+  stages {
+    stage("CI") {
+      steps { 
+        script {
+          def groovy_utilities = load "${pwd()}/groovy_utilities.groovy" 
+      
+          def cmakeFlags = [
+              "-DJP2KFLAG=ON",
+              "-DKAKADU_INCLUDE_DIR=/isisData/kakadu",
+              "-Dpybindings=OFF",
+              "-DCMAKE_BUILD_TYPE=RELEASE"
+          ]
+
+          def build_ok = true
+          def errors = []
+          def labels = ['CentOS', 'Fedora', 'Ubuntu', 'Mac'] // labels for Jenkins node types we will build on
+          def builders = [:] 
+          
+          println("Using Labels: " + labels)
+
+          for (x in labels) {
+            def label = x
+            def lower_label = label.toLowerCase()
+        
+            builders[lower_label] = {
+              node(lower_label) {
+              
+              def condaPath = ""
+              def isisEnv = [
+                "ISIS3DATA=/isisData/data",
+                "ISIS3TESTDATA=/isisData/testData",
+                "ISIS3MGRSCRIPTS=/isisData/data/isis3mgr_scripts",
+              ]
+              stage (label) {
+                  sh 'git config --global http.sslVerify false'
+                  checkout scm
+                  isisEnv.add("ISISROOT=${pwd()}/build")
+                  cmakeFlags.add("-DCMAKE_INSTALL_PREFIX=${pwd()}/install")
+                      
+                  env.STAGE_STATUS = "Creating conda environment"
+                  
+                  if (lower_label == "mac") {
+                    condaPath = "/tmp/" + sh(script: '{ date "+%m/%d/%y|%H:%M:%S:%m"; echo $WORKSPACE; } | md5 | tr -d "\n";', returnStdout: true) 
+                    
+                    sh """
+                      curl -o miniconda.sh  https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+                      bash miniconda.sh -b -p ${condaPath}
+                      """
+                  } else {
+                    condaPrefix = "/home/jenkins/.conda/envs/isis/"
+                    condaPath = "/home/jenkins/.conda/"
+                  } 
+           
+                  isisEnv.add("PATH=${pwd()}/install/bin:$condaPath/envs/isis/bin:$condaPath/bin:${env.PATH}")
+
+                  withEnv(isisEnv) {
+                    sh 'printenv'
+                    println("Anaconda Path: " + condaPath)
+                    
+                    sh """
+                        # Use the conda cache running on the Jenkins host
+                        conda config --set always_yes True
+                        conda config --set ssl_verify false 
+                    """
+                    
+                    if (lower_label == "centos") {
+                        sh "conda env create --prefix ${condaPath}/envs/isis -f environment_gcc4.yml"
+                    } else {
+                      sh """
+                          conda config --show channels
```

Comment:
Is this still needed? should it also be there for the centos environment?

Reply:
I had an issue where defaults were set above the channels in the env file. I think it is worth keeping for debug reasons as it doesn't hamper anything.